src/ResourceManager/Profile/Commands.Profile/Environment/AddAzureRMEnvironment.cs

```diff
@@ -112,12 +121,31 @@ protected override void BeginProcessing()
         public override void ExecuteCmdlet()
         {
             var profileClient = new RMProfileClient(AzureRmProfileProvider.Instance.Profile);
+            var result = profileClient.ListEnvironments(Name).Select(s => (PSAzureEnvironment)s).ToList();
+            if (result.Count != 0)
+            {
+                throw new InvalidOperationException("The specified environment already exists. Use a different name to create a new environment.");
+            }
 
             var newEnvironment = new AzureEnvironment
+                                 {
+                                     Name = Name,
+                                     OnPremise = EnableAdfsAuthentication
+                                 };
+
+            if (this.ParameterSetName.Equals(ArmParameterSet, StringComparison.Ordinal))
             {
-                Name = Name,
-                OnPremise = EnableAdfsAuthentication
-            };
+                MetadataResponse metadataEndpoints = new MetadataResponse();
+                metadataEndpoints = EnvironmentHelper.RetrieveMetaDataEndpoints(ResourceManagerEndpoint).Result;
+                string domain = EnvironmentHelper.RetrieveDomain(metadataEndpoints.PortalEndpoint);
+                ActiveDirectoryEndpoint = metadataEndpoints.authentication.LoginEndpoint.TrimEnd('/') + '/';
+                ActiveDirectoryServiceEndpointResourceId = metadataEndpoints.authentication.Audiences[0];
+                GalleryEndpoint = metadataEndpoints.GalleryEndpoint;
+                GraphEndpoint = metadataEndpoints.GraphEndpoint;
+                AzureKeyVaultDnsSuffix = $"vault.{domain}".ToLowerInvariant();
```

Comment:
This is using c#  6  features (string interpolation) which may not be available in the ci compiler.